node/node_test.go

```diff
@@ -0,0 +1,83 @@
+package node
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	proxy "github.com/cometbft/cometbft/proxy"
+	cmtypes "github.com/cometbft/cometbft/types"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/rollkit/rollkit/config"
+	test "github.com/rollkit/rollkit/test/log"
+	"github.com/rollkit/rollkit/types"
+)
+
+// cleanUpNode stops the node and checks if it is running
+func cleanUpNode(node Node, t *testing.T) {
+	defer func() {
+		if fn, ok := node.(*FullNode); ok {
+			fn.cancel()
+		}
+		if ln, ok := node.(*LightNode); ok {
+			ln.cancel()
+		}
+	}()
+	assert.NoError(t, node.Stop())
+	assert.False(t, node.IsRunning())
+}
+
+func initializeAndStartFullNode(ctx context.Context, t *testing.T) *FullNode {
+	node := initializeAndStartNode(ctx, t, "full")
+	return node.(*FullNode)
+}
+func initializeAndStartLightNode(ctx context.Context, t *testing.T) *LightNode {
+	node := initializeAndStartNode(ctx, t, "light")
+	return node.(*LightNode)
+}
+
+// initializeAndStartNode initializes and starts a test node
+func initializeAndStartNode(ctx context.Context, t *testing.T, nodeType string) Node {
+	node := setupTestNode(ctx, t, nodeType)
+	require.False(t, node.IsRunning())
+	err := node.Start()
+	require.NoError(t, err)
```

Comment:
Since we're using `require` here, the program will abort if `node.Start()` returns an error.  Do we still want to clean up node in that case?

Reply:
if the node hasn't started there isn't anything to clean up right? All we have done so far is to define structs which will just get dumped when the test exits.